zstd/tests/test.bats

```diff
@@ -0,0 +1,63 @@
+@test "package directory for package ident ${TEST_PKG_IDENT} exists" {
+  [ -d "/hab/pkgs/${TEST_PKG_IDENT}" ]
+}
+
+@test "zstd exe runs" {
+  run hab pkg exec $TEST_PKG_IDENT zstd
+  [ $status -eq 0 ]
+}
+
+expected_version="$(echo $TEST_PKG_IDENT | cut -d/ -f 3)"
+@test "zstd exe output mentions expected version $expected_version" {
+  actual_version="$(hab pkg exec "${TEST_PKG_IDENT}" zstd -h | head -1 | grep -o -E '[0-9]+\.[0-9]+\.[0-9]+')"
+  [ "$actual_version" = "$expected_version" ]
+}
+
+@test "zstd exe correctly linked" {
+  run ldd /hab/pkgs/$TEST_PKG_IDENT/bin/zstd
```

Comment:
This doesn't actually test correct linking. In the case a required library is not found, the exit code is still `0` from `ldd`:

```
# ldd /hab/pkgs/core/zstd/1.3.8/20190201183921/bin/zstd
	linux-vdso.so.1 (0x00007ffd4597d000)
	libpthread.so.0 => not found
	libc.so.6 => /hab/pkgs/core/glibc/2.27/20190115002733/lib/libc.so.6 (0x00007f64960a5000)
	/hab/pkgs/core/glibc/2.27/20190115002733/lib/ld-linux-x86-64.so.2 => /hab/pkgs/core/glibc/2.27/20190115002733/lib64/ld-linux-x86-64.so.2 (0x00007f6496261000)
```

(Note: Its linking fine, I removed `libpthread.so.0` to test ldd behaviour.)

Reply:
Nice catch, I didn't realize that. 

I think my options here are:
 * Keep these tests but re-word the descriptions to state that they load correctly (rather than they are linked).
 * Replace with file existence tests (which test less than ldd passing but cannot be confused with an invalid linking test)
 * Some other tests?

I would prefer to go with the first option unless you have some guidance here.

Reply:
Keeping the test would be testing only file existence and format. I don't think we need to test that.

The best alternative I think is based on @smacfarlane's suggestion to investigate use of https://github.com/habitat-sh/core-plans/blob/master/bin/check-bins.sh to check the binaries.

Right now I can't seem to locate his comment about that.

Reply:
I know what you're referring to, I put it in [an issue](https://github.com/habitat-sh/core-plans/issues/2628) so we'd not lose track of it. I think that will become a separate CI step than running these tests.

I think there's very little to test here then. I can just do a `test -f` on the expected library files existing.